contracts/UnderwriterToken.sol

```diff
@@ -27,8 +28,89 @@ import "@thesis/solidity-contracts/contracts/token/ERC20WithPermit.sol";
 ///         Anyone can submit this signature on the user's behalf by calling the
 ///         permit function, as specified in EIP2612 standard, paying gas fees,
 ///         and possibly performing other actions in the same transaction.
-contract UnderwriterToken is ERC20WithPermit {
+contract UnderwriterToken is ERC20WithPermit, Checkpoints {
```

Comment:
It also needs to override `beforeTokenTransfer` to write voting power and total supply checkpoints in case tokens were minted/burned.